### DISCUSSION TITLE
What do you expect from Entra PowerShell?

### DISCUSSION THREAD

**Discussion by alexandair** [microsoftgraph/entra-powershell#922] [General]
Here is my wish list:
- a proper pipeline support
- a meaningful output objects (no more `AdditionalProperties`)
- better default formatting that outputs useful properties and not just a bunch of IDs
- an easier way to find out required permissions (e.g. useful error messages that will tell exactly what permission is missing)
- a support for `-WhatIf` and `-Confirm` for commands that significantly change the state of Entra objects
- some kind of IntelliSense help for writing filters and search statements so that I don't need to guess what kind of operators are supported
- removing the need for `-CountVariable` when advanced query is needed (`-ConsistencyLevel` should be enough to redirect the request)

**Comment by DanielBradley1** [comment]
Automatic determination of an advanced query when filtering is a must! It would be such a time saver...

Also, a 'Show only properties with values' switch would be nice! I did a POC style thing for this where you could pipe the response > https://www.powershellgallery.com/packages/graph.refined/1.0/Content/graph.refined.ps1

**Comment by pminnebach** [comment]
Support for PowerShell native filters vs protocol. In a similar way as the AD cmdlets have -filter vs -ldapfilter.

**Comment by keithmweaver** [comment]
1. Users should not need to specify scopes when connecting, if a required scope is missing, it should prompt the user for the needed scope(s).
2. Entra PowerShell should have a dedicated first party enterprise application.
3. There should be a way for tenant admins to perform a one-time consent to all current & future scopes.

**Reply by SamErde** [reply]
> 1. Users should not need to specify scopes when connecting, if a required scope is missing, it should prompt the user for the needed scope(s).
> 
> 2. Entra PowerShell should have a dedicated first party enterprise application.
> 
> 3. There should be a way for tenant admins to perform a one-time consent to all current & future scopes.

I really like the first two, bit the third seems like it could create some risk for apps that are scoped differently for people or services with different roles.
